### DISCUSSION TITLE
Chapter 6: Trading system (core services)

### DISCUSSION THREAD

**Discussion by gvolpe** [gvolpe/trading#17] [General]
## Chapter 6: Trading system (core services)

⚠️ Note that the order and number of chapters is not yet fully decided, so expect changes here as well ⚠️

**Comment by PaNaVTEC** [comment]
Hi Gabriel, in the main diagram there's an arrow that is pointing in both directions (processor -> pulsar and pulsar -> processor) for the commands topic, shouldn't it be only consuming Commands and producing events? If that's the case maybe the arrow should be pointing from pulsar to the processor service only? At least that's what I get by reading the resources defined in `Main.scala` 

```
      trProducer <- Producer.pulsar[IO, TradeEvent](pulsar, evtTopic, evtSettings)
      swProducer <- Producer.pulsar[IO, SwitchEvent](pulsar, sweTopic, swtSettings)
      trConsumer <- Consumer.pulsar[IO, TradeCommand](pulsar, cmdTopic, cmdSub(config.appId))
      swConsumer <- Consumer.pulsar[IO, SwitchCommand](pulsar, swcTopic, swtSub(config.appId), compact)
```
Am I misunderstanding how the service works? Here's the extra arrow I'm asking:

![Screenshot from 2022-07-27 16-20-03](https://user-images.githubusercontent.com/837104/181272079-0c08f4b0-2dbc-48cc-be4f-b95bd58046ab.png)

Thanks!

**Reply by gvolpe** [reply]
Yes, that's an error in the diagram. Should be a unidirectional arrow, will be fixed in the next iteration, thanks!

**Reply by gvolpe** [reply]
For reference, this diagram appears first in Chapter 7 (Figure 7.0.1: overview).

**Reply by gvolpe** [reply]
Fixed both in the book (still unpublished) and the README file of the `trading` project (published).
